### PR TITLE
error serializer: serialize info and code too if present

### PR DESCRIPTION
**Pull request by go-oleg**
Building on top of https://github.com/juttle/juttle-jsdp/pull/8. Serialize info and code (used errors generated by juttle) as well if present.

@demmer @davidvgalbraith

### CODE REVIEW

File: test/protocol/serializers/error-serializer.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+describe('Error serializer', () => {
+    it('treats Error as serializable', () => {
+        expect(ErrorSerializer.isSerializable(new Error())).to.equal(true);
+    });
+
+    it('serializes and deserializes to Error', () => {
+        let err = new Error('this is the message');
+        err.info = { someInfo: 'a' };
+        err.code = 'SOME_ERROR';
+        expect(ErrorSerializer.deserialize(ErrorSerializer.serialize(err))).to.deep.equal(err);
```
Comment (davidvgalbraith) [inline test/protocol/serializers/error-serializer.js]: Need to check the actual properties here:

```
> try { 
    expect(new Error('expect sucks')).deep.equal(new Error('yeah')); 
    console.error('no error') 
} catch (err) { 
    console.error('whew it failed'); 
}
no error
```

Comment (go-oleg) [inline test/protocol/serializers/error-serializer.js]: Bah, `Error` is the worst. Will do, thx.

File: src/protocol/serializers/error-serializer.js [hunk 2]
```diff
@@ -4,10 +4,33 @@ const errorSerializer = {
...
+
+        if (value.info) {
+            result.info = value.info;
+        }
+
+        if (value.code) {
+            result.code = value.code;
+        }
+
+        return JSON.stringify(result);
```
Comment (davidvgalbraith) [inline src/protocol/serializers/error-serializer.js:19]: `JSON.stringify` blows away `undefined`s so in theory you could just `return JSON.stringify({message: value.message, code: value.code, info: value.info})`.

**Comment by davidvgalbraith** [comment]
+1

**Pull request merged**